logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/SetUserHandleUseCase.kt

```diff
@@ -19,19 +21,29 @@ sealed class SetUserHandleResult {
 }
 
 class SetUserHandleUseCase(
-    private val userRepository: UserRepository, private val validateUserHandleUseCase: ValidateUserHandleUseCase
+    private val userRepository: UserRepository,
+    private val validateUserHandleUseCase: ValidateUserHandleUseCase,
+    private val syncManager: SyncManager
 ) {
-    suspend operator fun invoke(handle: String): SetUserHandleResult = when (validateUserHandleUseCase(handle)) {
-        true -> userRepository.updateSelfHandle(handle).fold({
-            if (it is NetworkFailure.ServerMiscommunication && it.kaliumException is KaliumException.InvalidRequestError) {
-                handleSpecificError(it.kaliumException)
-            } else {
-                SetUserHandleResult.Failure.Generic(it)
-            }
-        }, {
-            SetUserHandleResult.Success
-        })
-        false -> SetUserHandleResult.Failure.InvalidHandle
+    suspend operator fun invoke(handle: String): SetUserHandleResult = suspending {
+        when (validateUserHandleUseCase(handle)) {
+            true -> userRepository.updateSelfHandle(handle).coFold(
+                {
+                    if (it is NetworkFailure.ServerMiscommunication && it.kaliumException is KaliumException.InvalidRequestError) {
+                        handleSpecificError(it.kaliumException)
+                    } else {
+                        SetUserHandleResult.Failure.Generic(it)
+                    }
+                }, {
+                    if (syncManager.isSlowSyncOngoing())
+                        syncManager.waitForSlowSyncToComplete()
```

Comment:
Can't this be moved out to the top like we do in other use cases?

Reply:
Ok so I moved it to the top, but I think `if (syncManager.isSlowSyncOngoing())` is worth keeping in this case because when we set the user handle right after the account creation, we don't have to start the sync and wait until it finishes, it can be done after the user handle is set. It's mainly for safety in case the user is already logged in and has not yet set the user handle so in the meantime the sync may have already been started.